spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/StackdriverJsonLayout.java

```diff
@@ -273,7 +273,7 @@ protected Map<String, Object> toJsonMap(ILoggingEvent event) {
           TimeUnit.MILLISECONDS.toSeconds(event.getTimeStamp()));
       map.put(
           StackdriverTraceConstants.TIMESTAMP_NANOS_ATTRIBUTE,
-          TimeUnit.MILLISECONDS.toNanos(event.getTimeStamp() % 1_000));
+          event.getNanoseconds());
```

Comment:
According to [documentation](https://logback.qos.ch/apidocs/ch/qos/logback/classic/spi/ILoggingEvent.html#getNanoseconds()), `getNanoseconds()` may return `-1`. We need to handle that case and default to the old method.

Reply:
Good point. Standard LoggingEvent from logback has it but there may be other implementations without it.